src/pages/AnalyticsPage/Chart/Chart.tsx

```diff
@@ -0,0 +1,137 @@
+import { format } from 'date-fns'
+import { Area, AreaChart, CartesianGrid, XAxis, YAxis } from 'recharts'
+
+import {
+  type ChartConfig,
+  ChartContainer,
+  ChartLegend,
+  ChartLegendContent,
+  ChartTooltip,
+  ChartTooltipContent,
+} from 'ui/Chart'
+import Icon from 'ui/Icon'
+import LoadingLogo from 'ui/LoadingLogo'
+
+import { useCoverage } from './useCoverage'
+
+// exporting for testing
+export const formatDate = (date: string) =>
+  format(new Date(date), 'MMM d, yyyy')
+
+interface ChartProps {
+  params: {
+    startDate: Date | null
+    endDate: Date | null
+    repositories: string[]
+  }
+}
+
+function Chart({ params }: ChartProps) {
+  const {
+    data: coverage,
+    isPreviousData,
+    isLoading,
+    isError,
+  } = useCoverage({
+    params: {
+      startDate: params.startDate ? params.startDate : null,
```

Comment:
Based on the type, we shouldn't need the ternary, right?

`startDate: params.startDate`

or even

`useCoverage({params, ...})`

Reply:
Done

Reply:
err gonna re-work these types a bit more